mailpoet/tests/performance/tests/subscribers-filtering.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-shadow */
+/* eslint-disable import/no-unresolved */
+/* eslint-disable import/no-default-export */
+/**
+ * External dependencies
+ */
+import { sleep, check, group } from 'k6';
+import { chromium } from 'k6/x/browser';
+import { randomIntBetween } from 'https://jslib.k6.io/k6-utils/1.1.0/index.js';
+
+/**
+ * Internal dependencies
+ */
+import {
+  baseURL,
+  thinkTimeMin,
+  thinkTimeMax,
+  headlessSet,
+  adminEmail,
+} from '../config.js';
+import { authenticate } from '../utils/helpers.js';
+/* global Promise */
+
+export function subscribersFiltering() {
+  const browser = chromium.launch({ headless: headlessSet });
+  const page = browser.newPage();
+
+  group('Subscribers - Filter subscribers', function subscribersFiltering() {
+    page
+      .goto(`${baseURL}/wp-admin/admin.php?page=mailpoet-subscribers`, {
+        waitUntil: 'networkidle',
+      })
+
+      .then(() => {
+        authenticate(page);
+      })
+
+      .then(() => {
+        return Promise.all([
+          page.waitForNavigation({ waitUntil: 'networkidle' }),
+        ]);
+      })
+
+      .then(() => {
+        page.locator('[data-automation-id="filters_subscribed"]').click();
+        page.waitForSelector('[data-automation-id="filters_subscribed"]');
+        check(page, {
+          'subscribers filter is visible': page
+            .locator('[data-automation-id="listing_filter_segment"]')
+            .isVisible(),
+        });
+      })
+
+      .then(() => {
+        page
+          .locator('[data-automation-id="listing_filter_segment"]')
+          .selectOption('3');
+        sleep(2);
+        page.waitForSelector('[data-automation-id="filters_subscribed"]');
+        check(page, {
+          'subscribers filter is visible': page
+            .locator('[data-automation-id="listing_filter_segment"]')
+            .isVisible(),
+        });
+      })
+
+      .then(() => {
+        page.waitForNavigation({ waitUntil: 'networkidle' });
+        page.locator('#search_input').type(adminEmail, { delay: 50 });
+        sleep(2);
+        page.waitForSelector('[data-automation-id="filters_subscribed"]');
+        check(page, {
+          'subscribers filter is visible': page
+            .locator('[data-automation-id="listing_filter_segment"]')
+            .isVisible(),
+        });
+      })
+
+      .finally(() => {
+        page.close();
+        browser.close();
+      });
+  });
+
+  sleep(randomIntBetween(`${thinkTimeMin}`, `${thinkTimeMax}`));
+}
+
+export default function subscribersFilteringTest() {
```

Comment:
I don't see `subscribersFilteringTest` used anywhere, but I do see `newsletterListing` used. Is this export getting used implicitly somehow, or is it necessary at all?

Reply:
The name of the function is required by Linter, can't be without a name and that's why I have to name all those functions and use `subscribersFiltering` in scenarios.

Reply:
What I meant is that I don't see this default export being referenced anywhere. In `scenarios.js` where you have this:

```js
import { subscribersFiltering } from './tests/subscribers-filtering.js';
```

you're importing a named import, i.e. the function that you define as an export on line 25 of `subscribers-filtering.js`:

```js
export function subscribersFiltering() {
```

If you were using the default export there wouldn't be the curly braces.

Long story short, does the code still work if you completely remove the default export?

Reply:
Good point. So if I remove the default export the code is working but then not executing it in `scenarios.js`.
I tried also adding `default` to the test function and it worked to execute it but again not in `scenarios.js`.

Reply:
I think I understand now. I didn't realize that if you want to run a specific test directly it depends on there being a default export.

It doesn't work to change it to `export default function subscribersFiltering() {...` because, I believe, it no longer creates a named export, and `scenarios.js` is explicitly importing named exports.

I think the best way to do this would be to keep the function defined the way it is now, `export function subscribersFiltering() {...`, and then for the default export, instead of creating a new function that just calls the first one, you can export the function itself as the default like this:

```js
export default subscribersFiltering;
```

This seemed to work for me anyway!

Reply:
I tried your suggestion but was unable to run the test in `scenarios.js`. It worked only by executing the individual test.
So to confirm, I tried with:
`export default subscribersFiltering;`
to be placed at the top bottom of the test and executing scenarios.
also tried with:
`export default subscribersFiltering();`
just in case, but didn't work.

I will continue as-is and if you feel it really worked for you to execute scenarios as well, please ping me on Slack to make a separate PR for that change. Thanks for very good review of the PR!